bitnami/postgresql-ha/templates/postgresql/scripts-configmap.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ printf "%s-scripts" (include "postgresql-ha.postgresql" .) }}
+  labels: {{- include "postgresql-ha.labels" . | nindent 4 }}
+    app.kubernetes.io/component: postgresql
+data:
+  pre-stop.sh: |-
+    #!/bin/bash
+
+    set -o errexit
+    set -o pipefail
+    set -o nounset
+
+    # Debug section
+    exec 3>&1
+    exec 4>&2
+
+    if [ "${BITNAMI_DEBUG}" = true ]; then
+      echo "==> Bash debug is on"
+    else
+      echo "==> Bash debug is off"
+      exec 1>/dev/null
+      exec 2>/dev/null
+    fi
+
+    # Load Libraries
+    . /opt/bitnami/scripts/libpostgresql.sh
+    . /opt/bitnami/scripts/librepmgr.sh
+
+    # Load PostgreSQL & repmgr environment variables
+    eval "$(repmgr_env)"
+    eval "$(postgresql_env)"
+
+    postgresql_enable_nss_wrapper
+
+    # Prepare env vars for managing roles
+    eval "$(repmgr_set_role)"
+
+    # Stop postgresql for graceful exit.
+    postgresql_stop
+
+    if [[ "$REPMGR_CURRENT_PRIMARY_HOST" == "$REPMGR_NODE_NETWORK_NAME" ]]; then
+      echo "Primary node need to wait for a new primary node before leaving the cluster"
+
+      wait="1"
+      while [ "$wait" -eq "1" ]
+      do
```

Comment:
Use `while [[ ]]; do` syntax

Reply:
I would borrow the `retry_while` function from https://github.com/bitnami/bitnami-docker-postgresql-repmgr/blob/master/11/debian-10/prebuildfs/opt/bitnami/scripts/libos.sh
It's more elegant

Reply:
I did a try using `retry_while` but it doesn't fit at all, we must wait for a condition, not for command execution. Also, I had to do sth like:
```
am_i_primary() {
  # use some libraries
  # check condition
}

retry_while "am_i_primary" 10 2
```
When using it inside `retry_while` the scope change and library functions don't work properly.

Reply:
You can wait for a condition using `retry_while`, you just create an auxiliary function, sth like `is_new_primary_ready` that returns "0/1" based on that condition. Then you can use:
```
    retry_while "is_new_primary_ready " 10 2
```
The scope shouldn't be an issue either. You should be able to access the same env. variables in the auxiliary function.

Reply:
I've tried again but it doesn't work

```
    is_new_primary_ready() {
        return_value=1
        eval "$(repmgr_set_role)"

        info "$REPMGR_CURRENT_PRIMARY_HOST == $REPMGR_NODE_NETWORK_NAME"
        if [[ "$REPMGR_CURRENT_PRIMARY_HOST" != "$REPMGR_NODE_NETWORK_NAME" ]]; then
            info "New primary detected, Leaving the cluster"
            return_value=0
        else
            info "I'm still primary node, waiting for a new primary"
        fi

        return $return_value
    }

    if [[ "$REPMGR_CURRENT_PRIMARY_HOST" == "$REPMGR_NODE_NETWORK_NAME" ]]; then
        info "Primary node need to wait for a new primary node before leaving the cluster"

        retry_while "is_new_primary_ready " 10 2
    else
        info "Standby node doesn't need to wait, leaving the cluster."
    fi
```
I would say this `repmgr_set_role` is missing sth inside `libos.sh`, I don't know but it works without `retry_while`

Reply:
`$REPMGR_CURRENT_PRIMARY_HOST` is not well resolved when using `retry_while`